tests/src/core/testqgsgdalprovider.cpp

```diff
@@ -647,7 +647,9 @@ void TestQgsGdalProvider::testGdalProviderQuerySublayersFastScan()
   QCOMPARE( res.at( 0 ).uri(), QStringLiteral( TEST_DATA_DIR ) + "/landsat.tif" );
   QCOMPARE( res.at( 0 ).providerKey(), QStringLiteral( "gdal" ) );
   QCOMPARE( res.at( 0 ).type(), QgsMapLayerType::RasterLayer );
-  QVERIFY( !res.at( 0 ).skippedContainerScan() );
+  // QVERIFY( !res.at( 0 ).skippedContainerScan() );
```

Comment:
this should be ifdefed based on gdal version, not completely removed

Reply:
I'd happily hand over to you to show how you would do this. see also #47983

Reply:
what do you think about this? inspired from your PR #44594
```
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,4,0)
+  QVERIFY( res.at( 0 ).skippedContainerScan() );
+#else
   QVERIFY( !res.at( 0 ).skippedContainerScan() );
+#endif
```